packages/cqrs/lib/src/cqrs.dart

```diff
@@ -166,4 +310,38 @@ class Cqrs {
       },
     ).timeout(_timeout);
   }
+
+  void _log(
+    CqrsMethod method,
+    _ResultType result, [
+    Object? error,
+    StackTrace? stackTrace,
+    List<ValidationError> validationErrors = const [],
+  ]) {
+    final log = switch (result) {
```

Comment:
I'd early return if logger is null. All that work is for nothing if you have no logger.